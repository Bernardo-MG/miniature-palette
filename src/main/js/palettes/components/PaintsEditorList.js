import React, { Fragment } from 'react';

import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import TextField from '@material-ui/core/TextField';

import AddCircleIcon from '@material-ui/icons/AddCircle';
import RemoveCircleOutlineIcon from '@material-ui/icons/RemoveCircleOutline';

function PaintsEditorList({ data, onAdd, onRemove, onChange, onBlur, errors, touched }) {
   let list;

   if (data && data.length > 0) {
      list = <Fragment>
         <Grid item xs={12}>
            <List>
               {data.map((paint, index) =>
                  <ListItem key={index}>
                     <ListItemText>
                        <TextField
                           fullWidth
                           id={`paints[${index}].name`}
                           name={`paints[${index}].name`}
                           label="paint_name"
                           value={paint.name}
                           onChange={onChange}
                           onBlur={onBlur}
                           helperText={(errors.name && touched.name) && errors.name}
                        />
                     </ListItemText>
                     <ListItemSecondaryAction>
                        <IconButton edge="end" aria-label="delete" onClick={() => onRemove(index)}>
                           <RemoveCircleOutlineIcon />
                        </IconButton>
                     </ListItemSecondaryAction>
                  </ListItem>
               )}
            </List>
         </Grid>
         <Grid item align="center" xs={12}>
            <IconButton aria-label="add" onClick={onAdd}>
               <AddCircleIcon />
            </IconButton>
         </Grid>
      </Fragment>;
   } else {
      list = <Grid item align="center" xs={12}>
         <IconButton aria-label="add" onClick={onAdd}>
            <AddCircleIcon />
         </IconButton>
      </Grid>;
   }

   return list;
}

PaintsEditorList.propTypes = {
   data: PropTypes.array.isRequired,
   onAdd: PropTypes.func.isRequired,
   onRemove: PropTypes.func.isRequired,
   onChange: PropTypes.func.isRequired,
   onBlur: PropTypes.func.isRequired,
   errors: PropTypes.array.isRequired
};

export default PaintsEditorList;
