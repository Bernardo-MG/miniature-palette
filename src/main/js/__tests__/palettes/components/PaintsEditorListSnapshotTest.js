import React from 'react'
import renderer from 'react-test-renderer';

import PaintsEditorList from 'palettes/components/PaintsEditorList';

describe('<PaintsEditorList />', () => {
   it('renders correctly when there is no data', () => {
      const tree = renderer
         .create(<PaintsEditorList
               data={ [] }
               onAdd={ () => 'action' }
               onRemove={ () => 'action' }
               onChange={ () => 'action' }
               onBlur={ () => 'action' }
               errors={ {} } />)
         .toJSON();
      expect(tree).toMatchSnapshot();
   }),
   it('renders correctly with data', () => {
      const tree = renderer
         .create(<PaintsEditorList
               data={ [{ name: 'name' }] }
               onAdd={ () => 'action' }
               onRemove={ () => 'action' }
               onChange={ () => 'action' }
               onBlur={ () => 'action' }
               errors={ {} } />)
         .toJSON();
      expect(tree).toMatchSnapshot();
   })
});
