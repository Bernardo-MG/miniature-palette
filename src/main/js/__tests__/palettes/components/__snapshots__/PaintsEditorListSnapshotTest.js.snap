// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaintsEditorList /> renders correctly when there is no data 1`] = `
<div
  align="center"
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
>
  <button
    aria-label="add"
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`<PaintsEditorList /> renders correctly with data 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <ul
      className="MuiList-root MuiList-padding"
    >
      <li
        className="MuiListItem-container"
      >
        <div
          className="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
          disabled={false}
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink={false}
                  htmlFor="paints[0].name"
                  id="paints[0].name-label"
                >
                  paint_name
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiInput-input"
                    disabled={false}
                    id="paints[0].name"
                    name="paints[0].name"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value="name"
                  />
                </div>
              </div>
            </span>
          </div>
        </div>
        <div
          className="MuiListItemSecondaryAction-root"
        >
          <button
            aria-label="delete"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
    </ul>
  </div>,
  <div
    align="center"
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <button
      aria-label="add"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;
